<launch>
    <arg name="sim" default="false" />
    <group unless="$(arg sim)">
        <include file="$(find sensors)/launch/depth_sensor.launch"/>
		<include file="$(find sensors)/launch/imu.launch"/>
		<include file="$(find sensors)/launch/dvl.launch"/>
		

		<param name="q_imunominal_imu_w" value="1.0" />
		<param name="q_imunominal_imu_x" value="0.0" />
		<param name="q_imunominal_imu_y" value="0.0" />
		<param name="q_imunominal_imu_z" value="0.0"/>

		<param name="q_dvl_auv_w" value="0.0" />
		<param name="q_dvl_auv_x" value="1.0" />
		<param name="q_dvl_auv_y" value="0.0"/>
		<param name="q_dvl_auv_z" value="0.0" />

		<param name="auv_dvl_offset_x" value="0.0" />
		<param name="auv_dvl_offset_y" value="0.0" />
		<param name="auv_dvl_offset_z" value="-0.3" />

		<param name="update_state_on_clock" value="false" />
		<param name="update_rate" value="100" />
    </group>
    <group if="$(arg sim)">
		<param name="update_state_on_clock" value="true" />
    </group>

	<param name="hydrophones_dx" value="0.1"/>
	<param name="hydrophones_dy" value="0.1"/>
	<node name="hydrophones_bearing"
			pkg="state_estimation"
			type="hydrophones_bearing.py"
			output="screen"
			respawn="true">
	</node>

	<node name="state_aggregator" pkg="state_estimation" type="state_aggregator.py" respawn="true" output="screen"/>
</launch>
